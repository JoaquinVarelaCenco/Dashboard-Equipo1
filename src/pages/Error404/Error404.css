.error404__Main{
    display: flex;
    flex-direction: column;
    text-align: center;
    height: calc(100vh - 72px);
    width: 100%;
}

.error404__top{
    display: flex;
    justify-content: center;
    align-items: flex-end;
    width: 100%;
    height: 50%;
    background-color: aliceblue;
}

.error404__top h2{
    margin-bottom: -75px;
    font-size: 350px;
    font-weight: bold;
    color:gainsboro;
}

.error404__bottom{
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 12px;
    padding: 50px;
    width: 100%;
    height: 50%;
    background-color: gainsboro;
}

.error404__bottom p{
    font-size: 35px;
}

.error404__bottom p:nth-child(2){
    font-size: 22px;
    text-transform: uppercase;
}

.error404__bottom button{
    background-color: #5b5d5d ;
    border: none;
    padding: 8px 10px;
    font-size: 16px;
    color: white;
    margin: 0 6px;
    border-radius: 15px;
    min-width: 150px;
}

@media (max-width: 690px){
    .error404__top h2{
        font-size: 200px;
        margin-bottom: -50px;
    }
    .error404__bottom p{
        font-size: 25px;
    }
    
    .error404__bottom p:nth-child(2){
        font-size: 16px;
        text-transform: uppercase;
    }
}

/* DarkMode */

.dark .error404__top{
    background-color: var(--dark-Background);
}

.dark .error404__bottom{
    background-color: var(--dark-BlackLight);
}

.dark .error404__top h2{
    color: var(--dark-BlackLight);
}