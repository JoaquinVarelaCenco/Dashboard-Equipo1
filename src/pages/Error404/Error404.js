import { Link } from 'react-router-dom';
import './Error404.css';

const Error404 = () => {
  return (
    <>
        <div className='error404__Main'>
            <div className='error404__top'>
                <p>404</p>
            </div>
            <div className='error404__bottom'>
                <h3>Disculpa, Página No Encontrada</h3>
                <p>La página que deseas acceder no existe</p>
                <Link to="/">
                    <button>Home</button>
                </Link>
            </div>
        </div>
    </>
  )
}

export default Error404