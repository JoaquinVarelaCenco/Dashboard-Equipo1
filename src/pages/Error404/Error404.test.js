import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Error404 from "./Error404";

let component;
let btnInicio;

describe('Test de la vista Error404', ()=>{

    beforeEach(()=>{
        component = render(<Error404 />, { wrapper: BrowserRouter })
        btnInicio = screen.getByRole('link');
    })

    it('La vista debe renderizarse correctamente', ()=>{
        const { container } = component;

        expect(container).toMatchSnapshot();
    });

    it('El boton debe contener la palabra inicio', ()=>{
        expect(btnInicio).toHaveTextContent(/inicio/i)
    });

    it('El botÃ³n debe redirigir a la ruta de Home', ()=>{
        expect(btnInicio.getAttribute('href')).toBe('/home')
    })

})