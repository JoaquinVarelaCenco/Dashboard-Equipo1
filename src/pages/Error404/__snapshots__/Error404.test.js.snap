// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de la vista Error404 La vista debe renderizarse correctamente 1`] = `
<div>
  <div
    class="error404__Main"
  >
    <div
      class="error404__top"
    >
      <h2>
        404
      </h2>
    </div>
    <div
      class="error404__bottom"
    >
      <p>
        Disculpa, Página No Encontrada
      </p>
      <p>
        La página que deseas acceder no existe
      </p>
      <a
        href="/home"
      >
        <button>
          Inicio
        </button>
      </a>
    </div>
  </div>
</div>
`;
